core/src/log.rs

```diff
@@ -140,3 +140,16 @@ pub trait ActionEvent {
 use tracing::Value;
 
 pub use crate::{debug, error, info, trace, warn};
+
+#[macro_export]
+macro_rules! bug_condition {
+    ($($arg:tt)*) => {{
+        if std::env::var("PANIC_ON_BUG")
```

Comment:
I would add an `OPENMINA_` prefix here and make sure we always use that in our env vars